src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Converters.cs

```diff
@@ -122,29 +125,71 @@ public JsonConverter GetConverter(Type typeToConvert)
             }
 
             DefaultJsonTypeInfoResolver.RootDefaultInstance();
-            return GetConverterInternal(typeToConvert);
+            return GetConverterFromTypeInfo(typeToConvert);
+        }
+
+        /// <summary>
+        /// Same as GetConverter but does not root converters
+        /// </summary>
+        internal JsonConverter GetConverterFromTypeInfo(Type typeToConvert)
+        {
+            if (_cachingContext == null)
+            {
+                if (_isLockedInstance)
+                {
+                    InitializeCachingContext();
```

Comment:
Why does this method need to initialize the caching context? Couldn't it just check if it's null or not?

Reply:
it does above, it's case where we could be using cache but we're not